src/test/java/org/fogbowcloud/manager/core/processors/FulfilledProcessorTest.java

```diff
@@ -99,355 +95,365 @@ public void tearDown() {
         Assert.assertNull(this.fulfilledOrderList.getNext());
     }
 
+    // test case: In calling the processFulfilledOrder() method for any order other than Fulfilled,
+    // you must not make state transition by keeping the order in your source list.
     @Test
     public void testProcessComputeOrderNotFulfilled()
             throws FogbowManagerException, UnexpectedException, InterruptedException {
-        
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FAILED);
         this.failedOrderList.addItem(order);
+        Assert.assertNull(this.fulfilledOrderList.getNext());
 
+        // exercise
         spyFulfiledProcessor();
-
         this.fulfilledProcessor.processFulfilledOrder(order);
+
+        // verify
+        Assert.assertNull(this.fulfilledOrderList.getNext());
         Assert.assertNotNull(this.failedOrderList.getNext());
     }
 
+    // test case: When running thread in the FulfilledProcessor without a LocalMember, the method
+    // processFulfilledOrder() must not change OrderState to Failed and must remain in Fulfilled
+    // list.
     @Test
     public void testRunProcessLocalComputeOrderWithoutLocalMember()
             throws FogbowManagerException, UnexpectedException, InterruptedException {
-        
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.failedOrderList.getNext());
 
         this.fulfilledProcessor = Mockito.spy(new FulfilledProcessor(REMOTE_MEMBER_ID,
                 this.tunnelingService, this.sshConnectivity,
                 DefaultConfigurationConstants.FULFILLED_ORDERS_SLEEP_TIME));
 
+        // exercise
         this.thread = new Thread(this.fulfilledProcessor);
         this.thread.start();
 
-        Thread.sleep(MIN_SLEEP_TIME);
-
+        // verify
         Assert.assertNotNull(this.fulfilledOrderList.getNext());
+        Assert.assertNull(this.failedOrderList.getNext());
     }
 
-    /**
-     * Test if a fulfilled order of an active localidentity compute instance has not being changed
-     * to failed if SSH connectivity is reachable.
-     *
-     * @throws InterruptedException
-     */
+    // test case: When running thread in the FulfilledProcessor and the InstanceState is Ready, the
+    // method processFulfilledOrder() must not change OrderState to Failed and must remain in
+    // Fulfilled list.
+    @SuppressWarnings("static-access")
     @Test
     public void testRunProcessLocalComputeOrderInstanceReachable() throws Exception {
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.failedOrderList.getNext());
 
         Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
```

Comment:
Do we really need to spy this object?

Reply:
No, I will remove the spy from this object.